### PR TITLE
Initial commit | Added base framework and Transformation related serv…

### PR DESCRIPTION
**Pull request by bharat-cd**
…ices and classes

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,43 @@
+plugins {
+    id 'org.springframework.boot' version '2.5.4'
+    id 'io.spring.dependency-management' version '1.0.11.RELEASE'
+    id 'java'
+}
+
+group = 'com.carrierdirect'
```
Comment (akumar-cd) [inline build.gradle:7]: I think group should be "com.carrierdirect.socket"

File: src/main/java/com/carrierdirect/SocketTransformationServicesApplication.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package com.carrierdirect;
```
Comment (akumar-cd) [inline src/main/java/com/carrierdirect/SocketTransformationServicesApplication.java:1]: I think we should have package "com.carrierdirect.socket.transformation"

File: src/main/java/com/carrierdirect/transformer/StringPropertyTransformer.java [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+
+/**
+ * {@link Transformer} implementation for String properties normalization
+ *
+ * @author Bharat
+ * @author CJ
+ * @author akumar
+ */
+@Slf4j
+public class StringPropertyTransformer extends AbstractPropertyTransformer {
```
Comment (akumar-cd) [inline src/main/java/com/carrierdirect/transformer/StringPropertyTransformer.java:27]: So we are creating all the core lib transformers within this service itself, as we will have only one transformation service?

**Review by akumar-cd — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (bharat-cd) [reply build.gradle:7]: Incorprated

File: src/main/java/com/carrierdirect/SocketTransformationServicesApplication.java (hunk 2 above)
Comment (bharat-cd) [reply src/main/java/com/carrierdirect/SocketTransformationServicesApplication.java:1]: Agree, incorporated

File: src/main/java/com/carrierdirect/transformer/StringPropertyTransformer.java (hunk 3 above)
Comment (bharat-cd) [reply src/main/java/com/carrierdirect/transformer/StringPropertyTransformer.java:27]: Thats correct

Comment (bharat-cd) [reply src/main/java/com/carrierdirect/transformer/StringPropertyTransformer.java]: By the way I have removed Address Contact Carrier related transformers as they are dependent on external utils, we can introduce them back as needed

Comment (akumar-cd) [reply src/main/java/com/carrierdirect/transformer/StringPropertyTransformer.java]: makes sense.

**Pull request merged**